### DISCUSSION TITLE
Option to save as gif or mp4

### DISCUSSION THREAD

**Discussion by Saortica** [Docile-Alligator/Infinity-For-Reddit#1609] [Ideas]
### Checklist

- [X] I have used the search function for [open](https://github.com/Docile-Alligator/Infinity-For-Reddit/discussions/categories/ideas) **and** [closed](https://github.com/Docile-Alligator/Infinity-For-Reddit/discussions/categories/completed-ideas) ideas to see if someone else has already submitted the same request.
- [X] I will describe the feature with as much detail as possible.

### Description

I would like the option (a toggle in settings) to download as either gif or mp4.

While gif is an annoying and inefficient filetype, the mp4 encodings are usually not the best and suffer some loss in detail. The option would allow the use case below. It could also been set by default to mp4 as of course this is likely the best settings for most users.

### Additional context

I've actually appreciated that Infinity saved as gif up until now, as I then have the option of encoding to mp4 with high quality settings with lame, if I want. This then results in the highest quality output if the original post was a gif (in many cases I've seen).

Realise it's a slightly niche use-case (many would not bother with this issue) but if not too much work to do, would increase flexibility of way the app handles video and assist those who like to retain max quality from the source. Thanks for the great app!

### How important is this feature?

None

**Comment by SqueakyMonjuer** [comment]
i support ts

**Reply by KindaSuS1368** [reply]
Me too, sometimes i see 100+ mb gifs that could be an mp4 and require only a few mbs
